Tests fail against `sympy` 1.10

Hi,

In maintaining the Nix package for `pydy`, we noticed that when we bumped our sympy version from 1.9 -> 1.10.1 `pydy` 0.6.0's tests started failing.

```
FAILED pydy/codegen/tests/test_c_code.py::TestCMatrixGenerator::test_generate_cse
FAILED pydy/codegen/tests/test_c_code.py::TestCMatrixGenerator::test_generate_code_blocks
FAILED pydy/codegen/tests/test_c_code.py::TestCMatrixGenerator::test_doprint
FAILED pydy/codegen/tests/test_octave_code.py::test_OctaveMatrixGenerator - A...
```

Full output is available at https://hydra.nixos.org/log/fsa3xzags44jyi27inh17f1s5q7pcshf-python3.9-pydy-0.6.0.drv. This happens on macos & linux, python 3.9 & 3.10.

It looks as though the tests are simply relying too much on very specific results coming from sympy:

```
>           assert g.doprint() == expected_m_file_1p2
E           AssertionError: assert 'function [ou...12];\n\nend\n' == 'function [ou...10];\n\nend\n'
E             Skipping 395 identical leading characters in diff, use -v to show
E             +  pydy_5 = -input_3(9);
E             -  pydy_5 = input_3(9).*pydy_4;
E             ?       ^
E             +     pydy_6 = input_3(9).*pydy_4;
E             ? +++      ^
E             -     pydy_6 = input_3(9) - pydy_3.*pydy_5;...
E             
E             ...Full output truncated (40 lines hidden), use '-vv' to show
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -35,8 +35,7 @@ jobs:
           miniforge-variant: Mambaforge
       - name: Install basic dependencies
         run: |
-          mamba install -q -y coverage cython matplotlib
-          mamba install -q -y nose numpy pythreejs sympy=1.8 scipy
+          mamba install -q -y coverage cython matplotlib nose numpy pythreejs sympy scipy
       - name: Test with nose
         run: nosetests -v --with-coverage --cover-package=pydy
       - name: Test installation of PyDy

--- a/pydy/codegen/tests/test_c_code.py
+++ b/pydy/codegen/tests/test_c_code.py
@@ -676,7 +676,81 @@ input_3[6] : [f0(t), f1(t), f2(t), f3(t), f4(t), f5(t)]
 
 */\
 """
-        if parse_version(SYMPY_VERSION) >= parse_version('1.2'):
+        if parse_version(SYMPY_VERSION) >= parse_version('1.10'):
+            expected_source = """\
+#include <math.h>
+#include "boogly_bee.h"
+
+void evaluate(
+              double input_0[19],
+              double input_1[6],
+              double input_2[6],
+              double input_3[6],
+              double output_0[36],
+              double output_1[6]
+             )
+{
+
+    double pydy_0 = input_0[17] + input_0[18];
+    double pydy_1 = input_0[16] + pydy_0;
+    double pydy_2 = input_0[15] + pydy_1;
+    double pydy_3 = input_0[14] + pydy_2;
+    double pydy_4 = input_3[5];
+    double pydy_5 = input_0[6] * input_0[17] + input_0[6] * input_0[18] +
+    pydy_4 + input_3[4];
+    double pydy_6 = input_0[6] * input_0[16] + pydy_5 + input_3[3];
+    double pydy_7 = input_0[6] * input_0[15] + pydy_6 + input_3[2];
+    double pydy_8 = input_0[6] * input_0[14] + pydy_7 + input_3[1];
+
+    output_0[0] = input_0[13] + pydy_3;
+    output_0[1] = pydy_3;
+    output_0[2] = pydy_2;
+    output_0[3] = pydy_1;
+    output_0[4] = pydy_0;
+    output_0[5] = input_0[18];
+    output_0[6] = pydy_3;
+    output_0[7] = pydy_3;
+    output_0[8] = pydy_2;
+    output_0[9] = pydy_1;
+    output_0[10] = pydy_0;
+    output_0[11] = input_0[18];
+    output_0[12] = pydy_2;
+    output_0[13] = pydy_2;
+    output_0[14] = pydy_2;
+    output_0[15] = pydy_1;
+    output_0[16] = pydy_0;
+    output_0[17] = input_0[18];
+    output_0[18] = pydy_1;
+    output_0[19] = pydy_1;
+    output_0[20] = pydy_1;
+    output_0[21] = pydy_1;
+    output_0[22] = pydy_0;
+    output_0[23] = input_0[18];
+    output_0[24] = pydy_0;
+    output_0[25] = pydy_0;
+    output_0[26] = pydy_0;
+    output_0[27] = pydy_0;
+    output_0[28] = pydy_0;
+    output_0[29] = input_0[18];
+    output_0[30] = input_0[18];
+    output_0[31] = input_0[18];
+    output_0[32] = input_0[18];
+    output_0[33] = input_0[18];
+    output_0[34] = input_0[18];
+    output_0[35] = input_0[18];
+
+    output_1[0] = -input_0[0] * input_2[0] + input_0[6] * input_0[13] -
+    input_0[7] * input_1[0] + pydy_8 + input_3[0];
+    output_1[1] = -input_0[1] * input_2[1] - input_0[8] * input_1[1] + pydy_8;
+    output_1[2] = -input_0[2] * input_2[2] - input_0[9] * input_1[2] + pydy_7;
+    output_1[3] = -input_0[3] * input_2[3] - input_0[10] * input_1[3] + pydy_6;
+    output_1[4] = -input_0[4] * input_2[4] - input_0[11] * input_1[4] + pydy_5;
+    output_1[5] = -input_0[5] * input_2[5] + input_0[6] * input_0[18] -
+    input_0[12] * input_1[5] + pydy_4;
+
+}\
+"""
+        elif parse_version(SYMPY_VERSION) >= parse_version('1.2'):
             expected_source = """\
 #include <math.h>
 #include "boogly_bee.h"

--- a/pydy/codegen/tests/test_c_code.py
+++ b/pydy/codegen/tests/test_c_code.py
@@ -271,7 +271,69 @@ input_1[6] : [x0(t), x1(t), x2(t), x3(t), x4(t), x5(t)]
 input_2[6] : [v0(t), v1(t), v2(t), v3(t), v4(t), v5(t)]
 input_3[6] : [f0(t), f1(t), f2(t), f3(t), f4(t), f5(t)]\
 """
-        if parse_version(SYMPY_VERSION) >= parse_version('1.2'):
+        if parse_version(SYMPY_VERSION) >= parse_version('1.10'):
+            expected['subexprs'] = \
+"""\
+    double pydy_0 = input_0[17] + input_0[18];
+    double pydy_1 = input_0[16] + pydy_0;
+    double pydy_2 = input_0[15] + pydy_1;
+    double pydy_3 = input_0[14] + pydy_2;
+    double pydy_4 = input_3[5];
+    double pydy_5 = input_0[6] * input_0[17] + input_0[6] * input_0[18] +
+    pydy_4 + input_3[4];
+    double pydy_6 = input_0[6] * input_0[16] + pydy_5 + input_3[3];
+    double pydy_7 = input_0[6] * input_0[15] + pydy_6 + input_3[2];
+    double pydy_8 = input_0[6] * input_0[14] + pydy_7 + input_3[1];\
+"""
+            expected['outputs'] = \
+"""\
+    output_0[0] = input_0[13] + pydy_3;
+    output_0[1] = pydy_3;
+    output_0[2] = pydy_2;
+    output_0[3] = pydy_1;
+    output_0[4] = pydy_0;
+    output_0[5] = input_0[18];
+    output_0[6] = pydy_3;
+    output_0[7] = pydy_3;
+    output_0[8] = pydy_2;
+    output_0[9] = pydy_1;
+    output_0[10] = pydy_0;
+    output_0[11] = input_0[18];
+    output_0[12] = pydy_2;
+    output_0[13] = pydy_2;
+    output_0[14] = pydy_2;
+    output_0[15] = pydy_1;
+    output_0[16] = pydy_0;
+    output_0[17] = input_0[18];
+    output_0[18] = pydy_1;
+    output_0[19] = pydy_1;
+    output_0[20] = pydy_1;
+    output_0[21] = pydy_1;
+    output_0[22] = pydy_0;
+    output_0[23] = input_0[18];
+    output_0[24] = pydy_0;
+    output_0[25] = pydy_0;
+    output_0[26] = pydy_0;
+    output_0[27] = pydy_0;
+    output_0[28] = pydy_0;
+    output_0[29] = input_0[18];
+    output_0[30] = input_0[18];
+    output_0[31] = input_0[18];
+    output_0[32] = input_0[18];
+    output_0[33] = input_0[18];
+    output_0[34] = input_0[18];
+    output_0[35] = input_0[18];
+
+    output_1[0] = -input_0[0] * input_2[0] + input_0[6] * input_0[13] -
+    input_0[7] * input_1[0] + pydy_8 + input_3[0];
+    output_1[1] = -input_0[1] * input_2[1] - input_0[8] * input_1[1] + pydy_8;
+    output_1[2] = -input_0[2] * input_2[2] - input_0[9] * input_1[2] + pydy_7;
+    output_1[3] = -input_0[3] * input_2[3] - input_0[10] * input_1[3] + pydy_6;
+    output_1[4] = -input_0[4] * input_2[4] - input_0[11] * input_1[4] + pydy_5;
+    output_1[5] = -input_0[5] * input_2[5] + input_0[6] * input_0[18] -
+    input_0[12] * input_1[5] + pydy_4;\
+"""
+        elif parse_version(SYMPY_VERSION) >= parse_version('1.2'):
             expected['subexprs'] = \
 """\
     double pydy_0 = input_0[17] + input_0[18];

--- a/pydy/codegen/tests/test_c_code.py
+++ b/pydy/codegen/tests/test_c_code.py
@@ -52,7 +52,31 @@ class TestCMatrixGenerator():
         v0, v1, v2, v3, v4, v5 = self.arguments[2]
         f0, f1, f2, f3, f4, f5 = self.arguments[3]
 
-        if parse_version(SYMPY_VERSION) >= parse_version('1.2'):
+        if parse_version(SYMPY_VERSION) >= parse_version('1.10'):
+            expected_subexprs = [
+                (pd[0], m4 + m5),
+                (pd[1], m3 + pd[0]),
+                (pd[2], m2 + pd[1]),
+                (pd[3], m1 + pd[2]),
+                (pd[4], f5),
+                (pd[5], g*m4 + g*m5 + pd[4] + f4),
+                (pd[6], g*m3 + pd[5] + f3),
+                (pd[7], g*m2 + pd[6] + f2),
+                (pd[8], g*m1 + pd[7] + f1)]
+            expected_simplified_matrices = (
+                sm.Matrix([[m0 + pd[3], pd[3], pd[2], pd[1], pd[0], m5],
+                           [pd[3], pd[3], pd[2], pd[1], pd[0], m5],
+                           [pd[2], pd[2], pd[2], pd[1], pd[0], m5],
+                           [pd[1], pd[1], pd[1], pd[1], pd[0], m5],
+                           [pd[0], pd[0], pd[0], pd[0], pd[0], m5],
+                           [m5,     m5,     m5,     m5,     m5, m5]]),
+                sm.Matrix([[-c0*v0 + g*m0 - k0*x0 + pd[8] + f0],
+                           [-c1*v1 - k1*x1 + pd[8]],
+                           [-c2*v2 - k2*x2 + pd[7]],
+                           [-c3*v3 - k3*x3 + pd[6]],
+                           [-c4*v4 - k4*x4 + pd[5]],
+                           [-c5*v5 + g*m5 - k5*x5 + pd[4]]]))
+        elif parse_version(SYMPY_VERSION) >= parse_version('1.2'):
             expected_subexprs = [
                 (pd[0], m4 + m5),
                 (pd[1], m3 + pd[0]),

--- a/pydy/codegen/tests/test_octave_code.py
+++ b/pydy/codegen/tests/test_octave_code.py
@@ -135,13 +135,47 @@ function [output_1] = eval_mats(input_1, input_2, input_3)
 end
 """
 
+    expected_m_file_1p10 = """\
+function [output_1] = eval_mats(input_1, input_2, input_3)
+% function [output_1] = eval_mats(input_1, input_2, input_3)
+%
+% input_1 : [x0(t), x1(t), x2(t)]
+% input_2 : [v0(t), v1(t), v2(t)]
+% input_3 : [c0, c1, c2, k0, k1, k2, m0, m1, m2]
+
+    pydy_0 = input_2(1);
+    pydy_1 = input_2(2);
+    pydy_2 = input_2(3);
+    pydy_3 = input_3(8) + input_3(9);
+    pydy_4 = 1./(input_3(7) + pydy_3);
+    pydy_5 = -input_3(9);
+    pydy_6 = input_3(9).*pydy_4;
+    pydy_7 = -pydy_3.*pydy_6 - pydy_5;
+    pydy_8 = 1./(-pydy_3.^2.*pydy_4 + pydy_3);
+    pydy_9 = input_3(1).*pydy_0 + input_3(4).*input_1(1);
+    pydy_10 = -pydy_9;
+    pydy_11 = input_3(2).*pydy_1 + input_3(5).*input_1(2) + ...
+    pydy_10.*pydy_3.*pydy_4;
+    pydy_12 = (-input_3(3).*pydy_2 - input_3(6).*input_1(3) - ...
+    pydy_10.*pydy_6 + pydy_11.*pydy_7.*pydy_8)./(-input_3(9).^2.*pydy_4 - ...
+    pydy_5 - pydy_7.^2.*pydy_8);
+    pydy_13 = pydy_8.*(-pydy_11 - pydy_12.*pydy_7);
+
+    output_1 = [pydy_0; pydy_1; pydy_2; pydy_4.*(-input_3(9).*pydy_12 - ...
+    pydy_13.*pydy_3 - pydy_9); pydy_13; pydy_12];
+
+end
+"""
+
     sys = multi_mass_spring_damper(3)
     q = sys.coordinates
     u = sys.speeds
     p = list(ordered(sys.constants_symbols))
     sym_rhs = sys.eom_method.rhs()
     g = OctaveMatrixGenerator([q, u, p], [sym_rhs])
-    if parse_version(SYMPY_VERSION) >= parse_version('1.2'):
+    if parse_version(SYMPY_VERSION) >= parse_version('1.10'):
+        assert g.doprint() == expected_m_file_1p10
+    elif parse_version(SYMPY_VERSION) >= parse_version('1.2'):
         assert g.doprint() == expected_m_file_1p2
     elif parse_version(SYMPY_VERSION) >= parse_version('1.1'):
         assert g.doprint() == expected_m_file_1p1

